orttraining/tools/amdgpu/Dockerfile.rocm4.2.pytorch

```diff
@@ -0,0 +1,175 @@
+  && cd ucx \
+  && git checkout v$UCX_VERSION \
+  && ./autogen.sh \
+  && mkdir build \
+  && cd build \
+  && ../contrib/configure-opt --prefix=$UCX_DIR --without-rocm --without-knem --without-cuda \
+  && make -j"$(nproc)" \
+  && make install \
+  && cd .. \
+  && rm -rf build
+
+# OpenMPI
+# note: require --enable-orterun-prefix-by-default for Azure machine learning compute
+# note: disable verbs as we use ucx middleware and don't want btl openib warnings
+WORKDIR $GITHUB_DIR
+ARG OPENMPI_BASEVERSION=4.0
+ARG OPENMPI_VERSION=${OPENMPI_BASEVERSION}.5
+ENV OPENMPI_DIR=$WORKSPACE_DIR/openmpi-${OPENMPI_VERSION}
+RUN git clone --recursive https://github.com/open-mpi/ompi.git \
+  && cd ompi \
+  && git checkout v$OPENMPI_VERSION \
+  && ./autogen.pl \
+  && mkdir build \
+  && cd build \
+  && ../configure --prefix=$OPENMPI_DIR --with-ucx=$UCX_DIR --without-verbs \
+                  --enable-mpirun-prefix-by-default --enable-orterun-prefix-by-default \
+                  --enable-mca-no-build=btl-uct --disable-mpi-fortran \
+  && make -j"$(nproc)" \
+  && make install \
+  && cd .. \
+  && rm -rf build \
+  && ldconfig \
+  && test -f ${OPENMPI_DIR}/bin/mpic++
+
+ENV PATH=$OPENMPI_DIR/bin:${PATH}
+ENV LD_LIBRARY_PATH=$OPENMPI_DIR/lib:${LD_LIBRARY_PATH}
+
+# Create a wrapper for OpenMPI to allow running as root by default
+RUN mv $OPENMPI_DIR/bin/mpirun $OPENMPI_DIR/bin/mpirun.real && \
+    echo '#!/bin/bash' > $OPENMPI_DIR/bin/mpirun && \
+    echo 'mpirun.real --allow-run-as-root "$@"' >> $OPENMPI_DIR/bin/mpirun && \
+    chmod a+x $OPENMPI_DIR/bin/mpirun
+
+# install mpi4py (be sure to link existing /opt/openmpi-xxx)
+RUN CC=mpicc MPICC=mpicc pip install mpi4py --no-binary mpi4py
+
+ARG CACHE_DATA=2021-05-18
+
+# ONNX Runtime
+WORKDIR $GITHUB_DIR
+ENV ORT_DIR=$GITHUB_DIR/onnxruntime
+RUN git clone --recursive https://github.com/microsoft/onnxruntime.git \
+  && cd onnxruntime \
+  && python3 tools/ci_build/build.py \
+    --cmake_extra_defines ONNXRUNTIME_VERSION=`cat ./VERSION_NUMBER` \
+    --build_dir build \
+    --config Release \
+    --parallel \
+    --skip_tests \
+    --build_wheel \
+    --use_rocm --rocm_home /opt/rocm \
+    --mpi_home $OPENMPI_DIR \
+    --nccl_home /opt/rocm \
+    --enable_training \
+  && test -f $ORT_DIR/build/Release/onnxruntime_training_bert \
+  && pip install $ORT_DIR/build/Release/dist/*.whl \
+  && ldconfig
+
+# ONNX Runtime Training Examples
+RUN pip3 install --no-cache-dir GPUtil azureml azureml-core datasets tokenizers ninja cerberus sympy sacremoses
+WORKDIR $GITHUB_DIR
+RUN git clone https://github.com/microsoft/huggingface-transformers.git
+
+ENV BERT_DIR=${WORKSPACE_DIR}/BERT
+ENV GPT2_DIR=${WORKSPACE_DIR}/GPT2
+ENV TRAIN_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.train.tokens
+ENV TEST_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.test.tokens
+
+# Enable ssh access without password needed
+RUN sed -i 's/#PermitRootLogin prohibit-password/PermitRootLogin yes/g' /etc/ssh/sshd_config
```

Comment:
Are these four lines needed?

Reply:
yeah, they should be removed as well. Thanks!

Reply:
But it is not changed?

Reply:
Ah, I was thinking you were referring the four lines below. Yeah, I was told that these four lines are needed.

+ENV BERT_DIR=${WORKSPACE_DIR}/BERT
+ENV GPT2_DIR=${WORKSPACE_DIR}/GPT2
+ENV TRAIN_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.train.tokens
+ENV TEST_FILE=${WORKSPACE_DIR}/GPT2/${GPT2_DATASET}/wiki.test.tokens